### DISCUSSION TITLE
Is there a Way I can check if a user is already paired with the authenticator app ?

### DISCUSSION THREAD

**Discussion by Muhsani95** [RobThree/TwoFactorAuth#136] [Q&A] [answered]
"I'm looking for a function to use in my app to check whether a user is already paired with the authenticator. In other words, I want to check if the username and secret are not already linked, and if they are not, display a QR code for pairing. If they are already linked, then perform another action."

**Comment by RobThree** [comment, marked as answer]
When you "pair" a user, you store the secret with the user. So if a user _has_ a secret, (s)he has 2FA enabled (or "paired"). If the user has no secret then (s)he hasn't enabled (or "paired") 2FA.

The QR you show is _always_ for pairing. You then confirm the code generated by the user's app by having them enter the TOTP code, and if that is correct you store the secret (which is the "pairing"). From then on, you never need to show a QR again, you just ask for the TOTP code on login each time.

Where / how you store the secret is up to you, but if your user is stored in, say, as `users` table then add a `totp_secret` field or something to the `users` table and have it nullable. Then, `null` means "not paired", else the field should contain [the shared secret](https://robthree.github.io/TwoFactorAuth/getting-started.html#3-shared-secrets) which means "paired" (and "2fa enabled").
